src/v/redpanda/admin_server.cc

```diff
@@ -463,6 +463,23 @@ ss::future<ss::httpd::redirect_exception> admin_server::redirect_to_leader(
       url, ss::httpd::reply::status_type::temporary_redirect);
 }
 
+namespace {
+bool need_redirect_to_leader(
+  model::ntp ntp, ss::sharded<cluster::metadata_cache>& metadata_cache) {
+    auto leader_id_opt = metadata_cache.local().get_leader_id(ntp);
+    if (!leader_id_opt.has_value()) {
+        throw ss::httpd::base_exception(
+          fmt_with_ctx(
```

Comment:
does this message get serialized back out to the user? if so, maybe we don't want to include file/line-numbers in the message?

Reply:
This code was just moved, but you're not wrong.  Changed to a plain fmt::format.